numpy/lib/index_tricks.py

```diff
@@ -769,8 +769,8 @@ def fill_diagonal(a, val, wrap=False):
     a : array, at least 2-D.
       Array whose diagonal is to be filled, it gets modified in-place.
 
-    val : scalar
-      Value to be written on the diagonal, its type must be compatible with
+    val : scalar or array_like optional
+      Scalar value or array representing the value to be written on the diagonal, its type must be compatible with
```

Comment:
Two comments here:
 1. The parameter description here should describe the behavior for both cases, i.e. when the value is a scalar, and when it is an array_like. The scalar case is obvious, but array_likes might be surprising: try `a = np.zeros((4, 5)); np.fill_diagonal(a, [1, 2])`, for example. There's no need to cover *every* single corner case here - the examples section is much better for illustrating details.
 2. Style nit - the lines should be kept at less than 80 characters per line.

Reply:
@rossbar I tried to address the the array parameter specifying that the length must be of the same length as the main diagonal. Not sure if the 80 characters per line is possible but definitely open to feedback on shortening it up!

Please take a look and do let me know :)